### PR TITLE
bug fixes

### PR DESCRIPTION
**Pull request by DrorT**
bug fixed:
1.  in walk state we checked if !reduction[key], but there might be cases when reduction[key] is 0, false or undefined and should stay like that
2. in make store operations reorganize the if statement, to skip non objects
3. in lifted reducer - reorganize the calls so that the special redux-operation INIT is called before redux INIT, allows other enhancer to clear the store when an the redux INIT is received to remove API specific data
4. fixes typo in error thrown

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -6,10 +6,9 @@ export const REDUX_OPERATION_SIGNATURE = '@@reduxOperations';
 
 export const walkState = (locationStack =[], state, initializer) => {
   return locationStack.reduce((reduction, key, currentIdx) => {
-    if (!reduction[key]) {
-      reduction[key] = (currentIdx === locationStack.length - 1) ? undefined : {};
-    }
-    return reduction[key] || initializer && initializer(undefined, {type:INIT_REDUX});
+    return reduction && reduction.hasOwnProperty(key) ? reduction[key] :
```
Comment (mattkrick) [inline src/index.js:9]: the initial value of reduction is state, so we're guaranteed reduction is going to an object, meaning we can get rid of the first check, right?

**Comment by DrorT** [comment]
Yeah

**Comment by DrorT** [comment]
Matt, 
My mistake the 1st check is essential as on init the reducer will get a state of undefined and this is exactly what we are looking for here.

**Comment by mattkrick** [comment]
makes sense, but in that case we should probably pull it out of the reduce method all together to clean up that code. I'll accept for now & make a todo for me to clean it up a little bit

**Pull request merged**